src/main/scala/loamstream/model/jobs/JobStatus.scala

```diff
@@ -14,7 +14,7 @@ sealed trait JobStatus {
   
   def isTerminal: Boolean
 
-  final def isFinished: Boolean = isSuccess || isFailure || isTerminal
+  final def isFinished: Boolean = isSuccess || isFailure || isTerminal || isCouldNotStart
```

Comment:
Can you define `isCouldNotStart`? Maybe me being pedantic, but it's ambiguous to me if it could not start now (i.e. for some dependency reason that will be resolved in the future such that it can start later) or if it is in this state because it will _never_ be able to start (e.g. some dependency failed)?

Reply:
It's the latter.  It means it can't start due to an upstream dependency failure.

Jobs that can't start (yet) are marked `NotStarted`.

However, `CouldNotStart` _is_ terminal (I suspect it wasn't always) so it doesn't need to be part of the test for `isFinished`.